### ISSUE TITLE
expected behavior

### ISSUE THREAD

**Issue by appsird** [insidegui/NoteTaker#2]
Ok, so I compiled the project for iOS and am running on my 5S. It shows a screen with blank table (below)
![img_0646 png](https://user-images.githubusercontent.com/8471833/30253844-1412a6da-9642-11e7-88ae-c8597f1a79c1.png)

I then launch the macOS version, and create some notes and I see the below - though still the iOS app is not synced. 

![x](https://user-images.githubusercontent.com/8471833/30253846-3afa705c-9642-11e7-949c-9052451917ba.png)

I looked at my development CK dashboard and I see only User record, no other record.

Is there some setup required in the dashboard, i.e. do I need to create the record type. Suspect the app should create on demand.

thanks for your assistance.

**Comment by appsird** [comment]
dashboard below:

![untitled](https://user-images.githubusercontent.com/8471833/30253860-8fe8be0c-9642-11e7-8889-c7756909b0bf.png)

**Comment by appsird** [comment]
In NoteSyncEngine.swfit I also set as follows:

static let noteTakerShared = CKContainer(identifier: "iCloud.net.appsird.NoteTakerShared")

For both NoteTaker and MobileNoteTaker targets I set the iCloud container to this custom container only. In my dashboard the container is created, though empty.

Bundle Identifier for NoteTaker  is net.appsird.NoteTaker 
Bundle Identifier for MobileNoteTaker is  net.appsird.MobileNoteTaker 

This is similar to the way the demo app was set.

It appears I error out in NoteSyncEngine.swift fetchCloudKitNotes->queryCompletionBlock and Subscriptions are not written.

I also receive the following warning which I suspect are OK:

warning: Swift error in module RealmSwift.
Debug info from this module will be unavailable in the debugger.

warning: Swift error in module RxSwift.
Debug info from this module will be unavailable in the debugger.

warning: Swift error in module RxBlocking.
Debug info from this module will be unavailable in the debugger.

warning: Swift error in module RxCocoa.
Debug info from this module will be unavailable in the debugger.

warning: Swift error in module RxOptional.
Debug info from this module will be unavailable in the debugger.

warning: Swift error in module RxRealm.
Debug info from this module will be unavailable in the debugger.

___

I admire your efforts, and it could really could beneficial to understand, though I seem to be a bit stuck.

Any assistance would be really appreciated.

**Comment by appsird** [comment]
Ended up I had various failures for several issues. I ended up placing the following everywhere in NoteSyncEngine where error returned nil. This allowed easier debug:

`assert(false, "\(error!.localizedDescription)")`

I currently error out in NoteSyncEngine.swift fetchCloudKitNotes->queryCompletionBlock
with message:

assertion failed: Unknown field 'modifiedAt': file /Users/brianx/Documents/NoteTaker-master/NoteTakerCore/Source/NotesSyncEngine.swift, line 96

Though, this field exists on my Note record - as it is created by the dashboard. I marked this field Queryable. Will continue to investigate. Seems the initial setting of the database is paramount to the example working.
